docs/articles_en/documentation/openvino-extensibility/transformation-api/patterns-python-api.rst

```diff
@@ -0,0 +1,180 @@
+This is why matching the same pattern against different nodes becomes possible. Basically, we may think of ``WrapType`` as "one of listed". Note, that you may provide more than 2 types
+to ``WrapType``.
+
+If you want to have some additional checking for you node, you may create a predicate for it providing a function or a lambda. This function will be executed during
+matching performing the additional validation specified in the logic of the function. For example, you may want to check the consumers count of a given node:
+
+.. doxygensnippet:: docs/snippets/ov_patterns.py
+   :language: python
+   :fragment: [ov:wrap_type_predicate]
+
+.. doxygensnippet:: docs/ov_patterns.cpp
+   :language: cpp
+   :fragment: [ov:wrap_type_predicate]
+
+2. AnyInput 
+++++++++++++++++++++++++++++++++++++++++
+You have already seen ``AnyInput`` in the above examples. We use it when we don't really care about a specific input for a given node.
+
+.. doxygensnippet:: docs/snippets/ov_patterns.py
+   :language: python
+   :fragment: [ov:any_input]
+
+.. doxygensnippet:: docs/ov_patterns.cpp
+   :language: cpp
+   :fragment: [ov:any_input]
+
+You may also create ``AnyInput()`` with a predicate, if you want some additional checks for you input. It would look similar to ``WrapType`` with a lambda or a function. Let's say we want to make sure the inputs has a rank of 4.
+
+.. doxygensnippet:: docs/snippets/ov_patterns.py
+   :language: python
+   :fragment: [ov:any_input_predicate]
+
+.. doxygensnippet:: docs/ov_patterns.cpp
+   :language: cpp
+   :fragment: [ov:any_input_predicate]
+
+3. Or
+++++++++++++++++++++++++++++++++++++++++
+``Or`` is somewhat similar to ``WrapType``, however if ``WrapType`` can only match one of types provided in the list, ``Or`` is used to match different _branches_ of nodes.
+It would be much easier to understand with a visualization. Let's say, we want to try to match the model against two different sequences of nodes. The ``Or`` type
+facilitates this by creating 2 different branches (``Or`` supports more than 2 branches). It would look as following:
+
+.. image:: ./../../../_static/images/or_branches.png
+
+As you may see, the red branch will not match, however it will work perfectly fine for the blue one.
+That's what it would look in code:
+
+.. doxygensnippet:: docs/snippets/ov_patterns.py
+   :language: python
+   :fragment: [ov:pattern_or]
+
+.. doxygensnippet:: docs/ov_patterns.cpp
+   :language: cpp
+   :fragment: [ov:pattern_or]
+
+Note that matching will succeed for the first matching branch and the remaining ones will not be checked.
+
+4. Optional
+++++++++++++++++++++++++++++++++++++++++
+``Optional`` is a bit tricky one. It allows to specify what node might be or might not present in the model. Under the hood
+the pattern will create 2 branches using ``Or``: one with the optional node present, another one without it. That's what it would look like visually with the ``Optional``
+unfolding into 2 branches:
```

Comment:
```suggestion
``Optional`` is a bit tricky. It allows specifying whether a node might be present or absent in the model. Under the hood,
the pattern will create two branches using ``Or``: one with the optional node present and another one without it. Here is what it would look like with the ``Optional``
unfolding into two branches:
```

Reply:
Fixed.